ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -327,11 +327,16 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			} else {
 				// We only need to delete the RayCluster. We don't need to delete the submitter Kubernetes Job so that users can still access
 				// the driver logs. In addition, a completed Kubernetes Job does not actually use any compute resources.
+
 				if _, err = r.deleteClusterResources(ctx, rayJobInstance); err != nil {
 					return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 				}
+				if err = r.Client.Delete(ctx, rayJobInstance); err != nil {
```

Comment:
Personally, I don't see a lot of benefit from deleting the RayJob itself as long as the underlying cluster resources are deleted. The RayJob and Job submission contains the job status and logs which is useful for debugging faliures without using any extra resources.

Reply:
If there's a real use-case for this behavior, can we consider a separate field so users can still delete the RayCluster without deleting the RayJob after completion?

Reply:
I also don't see the value of this behavior, but K8s Job has a similar behavior. If users set `ttlSecondsAfterFinished` in a Kubernetes Job, it will be garbage collected after completion. Since the Kubernetes Job exhibits similar behavior, I am okay with adding this feature as optional if the code changes are not extensive. What do you think? cc @andrewsykim

Reply:
I was actually going back to the original issue I opened for this.

I see value for cleaning up old jobs after a certain amount of time. Similar to K8s jobs. I think having this as optional is the correct move. @kevin85421 

@Xiao75896453 let me know if you want help or want to split up the work.

Reply:
> I also don't see the value of this behavior, but K8s Job has a similar behavior. If users set ttlSecondsAfterFinished in a Kubernetes Job, it will be garbage collected after completion. Since the Kubernetes Job exhibits similar behavior, I am okay with adding this feature as optional if the code changes are not extensive. What do you think? 

makes sense to me